tianshou/data/batch.py

```diff
@@ -376,8 +376,9 @@ def __setitem__(self, index: Union[
             except KeyError:
                 if isinstance(val, Batch):
                     self.__dict__[key][index] = Batch()
-                elif isinstance(val, np.ndarray) and \
-                        issubclass(val.dtype.type, (np.bool_, np.number)):
+                elif isinstance(val, torch.Tensor) or \
```

Comment:
I noticed this yesterday but I didn't care to fix it x) Nice thing that you did it !

Reply:
The exception comes up when I write MARL scripts, so I fixed it by the way.